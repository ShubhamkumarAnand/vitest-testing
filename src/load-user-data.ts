interface User {
	username: string;
	name: string;
	projects: string[];
	coolness?: number;
	favoriteFood: string;
}

async function loadUser(username: string): Promise<User | undefined> {
	const users: User[] = [
		{
			username: 'imskanand',
			name: 'shubham',
			projects: ['buyzon', 'hindu-way'],
			favoriteFood: 'litti-chokha',
		},
		{
			username: 'satyamjee',
			name: 'satyam',
			projects: ['love', 'laugh'],
			favoriteFood: 'litti-chokha',
		},
	];

	return users.find((user) => user.username === username);
}

export async function loadUserData(username: string) {
	const user = await loadUser(username);
	if (!user) {
		throw new Error('No User Found');
	}
	user.coolness = username === 'imskanand' ? 100 : -1;
	return user;
}
