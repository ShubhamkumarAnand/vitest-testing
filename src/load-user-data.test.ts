import { loadUserData } from './load-user-data';
import { describe, test, expect } from 'vitest';

describe('loadUserData', () => {
	test('load user data as expected', async () => {
		const user = await loadUserData('imskanand');
		expect(user).toMatchSnapshot();
	});

	test('Setting up the coolness correctly', async () => {
		const imskanand = await loadUserData('imskanand');
		const satyam = await loadUserData('satyamjee');

		expect(imskanand.coolness).toBe(100);
		expect(satyam.coolness).toBe(-1);
	});

	test('Throws the error which does not exists', async () => {
		expect(async () => await loadUserData('fakeUser')).rejects.toThrowError('No User Found');
	});
});
