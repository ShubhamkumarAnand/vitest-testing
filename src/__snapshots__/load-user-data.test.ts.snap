// Vitest Snapshot v1

exports[`loadUserData > load user data as expected 1`] = `
{
  "coolness": 100,
  "favoriteFood": "litti-chokha",
  "name": "shubham",
  "projects": [
    "buyzon",
    "hindu-way",
  ],
  "username": "imskanand",
}
`;
